wordpress-mobile/WordPress-Android
# Post settings excerpt

Fixes #5807. This is first of many post settings PRs. I wanted to start with something isolated, so picked the excerpt. The design is not fully implemented yet, this PR only handles the functionality of the excerpt as well as the excerpt dialog's design.

We might refactor a few things in the code to make them reusable, however I didn't want to over-engineer it. Those refactors will come when we need them in the following PRs, as well as a full design revamp of the post settings page. Feel free to ping me to ask if something's planned for further PRs to make sure I didn't miss anything. Thanks!

The excerpt is at the top of the page to make it easier to test, it'll move back to where it belongs in the further PRs.

To test:
* Go into the post settings page, tap on the excerpt and change it.
* After you change it, tap on it again, to make sure the dialog shows the updated text
* Remove the excerpt and make sure it's set to "Not Set"
* When the excerpt was removed, update the settings and make sure the excerpt on the remote is empty and not "Not Set"
